<!--Processes the whole state of the board.
It identifies the state of each cell: red, blue, or empty.
-->
<launch>
    <include file="$(find ttt_board_sensor)/launch/usb_cam_node.launch" />

    <!-- xml file containing the cells of the board -->
    <param name="baxter_tictactoe/board_file" textfile="$(find ttt_board_sensor)/resources/ttt_board_lab_B.xml" />

    <!-- Range thresholds for the red tokens -->
    <param name="baxter_tictactoe/h_low_red" type="int" value="165" />
    <param name="baxter_tictactoe/s_low_red" type="int" value="100" />
    <param name="baxter_tictactoe/v_low_red" type="int" value="0" />
    <param name="baxter_tictactoe/h_high_red" type="int" value="180" />
    <param name="baxter_tictactoe/s_high_red" type="int" value="256" />
    <param name="baxter_tictactoe/v_high_red" type="int" value="256" />

    <!-- Range thresholds for the blue tokens -->
    <param name="baxter_tictactoe/h_low_blue" type="int" value="90" />
    <param name="baxter_tictactoe/s_low_blue" type="int" value="0" />
    <param name="baxter_tictactoe/v_low_blue" type="int" value="0" />
    <param name="baxter_tictactoe/h_high_blue" type="int" value="130" />
    <param name="baxter_tictactoe/s_high_blue" type="int" value="256" />
    <param name="baxter_tictactoe/v_high_blue" type="int" value="256" />

    <!-- The colored area threshold specifies the minimum are size for which a cell is considered colored. -->
    <!-- It depends on the distance between the camera and board. -->
    <param name="baxter_tictactoe/color_area_threshold" type="int" value="100" />

    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find ttt_board_sensor)/custom_rosconsole.conf"/>

    <node name="board_state_sensor" pkg="ttt_board_sensor" type="board_state_sensing" respawn="false" output="screen" required="true">
        <remap from="/image_in" to="/usb_cam/image_raw"/>
    </node>

</launch>
